prov/sockets/src/sock_progress.c

```diff
@@ -2089,6 +2098,20 @@ static int sock_pe_progress_tx_entry(struct sock_pe *pe,
 	int ret;
 	struct sock_conn *conn = pe_entry->conn;
 
+	if (sock_comm_is_disconnected(pe_entry)) {
```

Comment:
This won't work as the method checks if ```pe_entry->conn->connected``` is reset and we only reset when system call ```recv()``` returns 0 meaning the socket is disconnected.

Sending data uses ```ofi_write_socket``` which calls ```write()``` system call. I am not sure if we can mark the socket disconnected if ```write()``` returns ```EPIPE```. @shefty, do you have any thoughts?